crates/biome_js_analyze/src/lint/nursery/no_static_element_id.rs

```diff
@@ -0,0 +1,121 @@
+use biome_analyze::{Rule, RuleDiagnostic, RuleDomain, context::RuleContext, declare_lint_rule};
+use biome_console::markup;
+use biome_diagnostics::Severity;
+use biome_js_semantic::SemanticModel;
+use biome_js_syntax::{
+    AnyJsExpression, AnyJsxAttributeValue, JsCallExpression, JsPropertyObjectMember, JsxAttribute,
+    jsx_ext::AnyJsxElement,
+};
+use biome_rowan::{AstNode, declare_node_union};
+
+use crate::react::{ReactApiCall, ReactCreateElementCall};
+use crate::services::semantic::Semantic;
+
+declare_lint_rule! {
+    /// Prevent the usage of static string literal `id` attribute on elements.
+    ///
+    /// In React, hardcoding IDs is discouraged because IDs have to be unique in the DOM.
+    /// You should use [`useId`](https://react.dev/reference/react/useId) to generate unique IDs for accessibility purposes.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <div id="foo">bar</div>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// React.createElement("div", { id: "foo" });
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// const id = useId();
+    /// <div id={id}>bar</div>;
+    /// ```
+    ///
+    /// ```jsx
+    /// const id = useId();
+    /// React.createElement("div", { id });
+    /// ```
+    ///
+    pub NoStaticElementId {
+        version: "next",
+        name: "noStaticElementId",
+        language: "jsx",
+        recommended: true,
```

Comment:
Let's make it false now
```suggestion
        recommended: false,
```